src/macaron/repo_finder/repo_validator.py

```diff
@@ -13,6 +13,8 @@
 def find_valid_repository_url(urls: Iterable[str]) -> str:
     """Find a valid URL from the provided URLs.
 
+    URLs are assumed to be a priority list, with the first match preventing further processing.
```

Comment:
I don't think we need to mention that this function accepted a priority url list because the implementation of this function doesn't care what "priority" would mean for `urls`. It's up to the callers to ensure what they provide has been ordered correctly. 
I think we can just say that this function will return the first valid repository URL from `urls` or something similar.